marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mastering/default/merging.sjs

```diff
@@ -40,84 +40,105 @@ function main(content, options) {
   const jobID = datahub.flow.globalContext.jobId;
   const urisPathReference = cts.pathReference('/matchSummary/URIsToProcess', ['type=string','collation=http://marklogic.com/collation/']);
   const datahubCreatedOnRef = cts.fieldReference('datahubCreatedOn', ['type=dateTime']);
-  const uriToTakeActionOn = content.uri;
+  const thisMatchSummaryURI = content.uri;
   masteringStepLib.checkOptions(null, options, null, requiredOptionProperties);
   const matchSummaryCollection = `datahubMasteringMatchSummary${options.targetEntityType ? `-${options.targetEntityType}`:''}`;
   const collectionQuery = cts.collectionQuery(matchSummaryCollection);
-  const uriRangeQuery = cts.rangeQuery(urisPathReference, '=', uriToTakeActionOn);
-  let relatedMatchSummaries = cts.search(
-    cts.andQuery([
-      uriRangeQuery,
-      collectionQuery
-    ]),
-    ['unfiltered',cts.indexOrder(datahubCreatedOnRef, 'descending')]
-  );
-  let mostRecentMatchSummary = fn.head(relatedMatchSummaries);
-  let matchSummaryJson = mostRecentMatchSummary.toObject();
-  let uriActionDetails = matchSummaryJson.matchSummary.actionDetails[uriToTakeActionOn];
-  // If the action is merge, ensure we create the merge with the most URIs
-  if (uriActionDetails && uriActionDetails.action === 'merge') {
-    const urisQuery = cts.pathRangeQuery('//actionDetails/*/uris', '=', uriActionDetails.uris)
-    const postiveQuery = [
-      cts.jsonPropertyValueQuery('action', 'merge'),
-      urisQuery,
-      collectionQuery
-    ];
-    const otherMergesForURI = cts.search(
-      cts.andNotQuery(
-        cts.andQuery(postiveQuery),
-        uriRangeQuery
-      ),
-      ['unfiltered',cts.indexOrder(datahubCreatedOnRef, 'descending')]
-    ).toArray()
-        // get the actionNode for the URI
-        .map(
-          (result) => result.xpath(`/matchSummary/actionDetails/*[action = 'merge']`).toArray()
-            .filter((actionNode) => cts.contains(actionNode, urisQuery))[0]
-        )
-        // filter out false positives that don't have merge actions for the given URI
-        .filter((actionNode) => !!actionNode)
-        // convert node to JSON object
-        .map((actionNode) => actionNode.toObject());
-    if (otherMergesForURI.some((otherMerge) => otherMerge.uris.length > uriActionDetails.uris.length)) {
-      return Sequence.from([]);
+
+  let returnArray = [];
+
+  let thisMatchSummary = fn.head(cts.search(cts.documentQuery(thisMatchSummaryURI), ['unfiltered']));
```

Comment:
You should be able to retrieve this from `content.value`.  Also in a case like this where we want a single document from a URI we can use `cts.doc("uri")`.